### DISCUSSION TITLE
Filenames

### DISCUSSION THREAD

**Discussion by robinkoehler** [croneter/PlexKodiConnect#1627] [Q&A] [answered]
Hi Cron and everyone, 

I have a question regarding the filenames that get passed on from Kodi / PKC.

I am using PotPlayer set up as my primary external video player in Kodi. I am using "profiles" within PotPlayer to change settings based on file name. 

If I load my files directly from explorer, all the profiles get applied correctly. If I load my files from Kodi, PotPlayer only loads the default profile. 

I have examined the filename that is getting passed on from Kodi, and it is just a URL with a few strings of seemingly random numbers. No actual file name, no extension. 

Is there a setting I can change within PKC, or is this an issue with Plex, or an issue with Kodi?

My desired outcome is for the URL that is passed on to PotPlayer to contain the actual file name so that my automated profile switching works. 

Thank you in advance!

Best wishes, 

Robin

Here is an example filename as read by PotPlayer:
General
Unique ID                      : 44662174079916087039622360584739926351 (0x21999F312E5EF1AEB7304227DC3CB14F)
Complete name                  : https://192-168-2-3.xxxxxxxxxxxxxxxxxxxxxxxxxx.plex.direct:32400/library/parts/19399/1607697353/

**Comment by piejanssens** [comment, marked as answer]
https://github.com/croneter/PlexKodiConnect/wiki/Direct-Paths-Explained

**Comment by robinkoehler** [comment]
> https://github.com/croneter/PlexKodiConnect/wiki/Direct-Paths-Explained

Hey piejanssens, 

That documentation got me there, thank you very much! Didn't know what to look for. 

Take care and best wishes, 

Robin
